tests/dotnet-sos/sos_test.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+import lldb
+import sys
+
+assembly = "hello.dll"
+function = "hello.Program.Main"
+
+def run_test(debugger):
+    debugger.SetAsync(False)
+    res = lldb.SBCommandReturnObject()
+    ci = debugger.GetCommandInterpreter()
+
+    debugger.HandleCommand("run")
+
+    process = debugger.GetSelectedTarget().GetProcess()
+
+    # We hit the cpuid SIGILL pretty early in dotnet initialization
+    assert(process.GetState() == lldb.eStateStopped)
+    assert(process.GetSelectedThread().GetStopReason() == lldb.eStopReasonSignal)
+    debugger.HandleCommand("pro hand -p true -s false -n false SIGILL")
+
+    # Set managed breakpoint
+    ci.HandleCommand("bpmd hello.dll hello.Program.Main", res)
+    assert(res.Succeeded())
+    process.Continue()
+
+    # Should've hit managed breakpoint
+    assert(process.GetState() == lldb.eStateStopped)
+    assert(process.GetSelectedThread().GetStopReason() == lldb.eStopReasonBreakpoint)
+
+    print("\nRun clrstack at breakpoint hit: ")
+    print("=================================")
+    ci.HandleCommand("clrstack", res)
```

Comment:
If we can capture the output of HandleCommand, then we can add assertions for frames

Reply:
For SOS commands, couldn't capture output via `res.GetOutput`, the command output goes directly to stdout. I have added grep assertions on the stdout as an alternative.